ChronoStore/src/storywriter.cpp

```diff
@@ -0,0 +1,497 @@
+#include <iostream>
+
+#define H5_SIZEOF_SSIZE_T H5_SIZEOF_LONG_LONG
+
+#include <hdf5.h>
+#include <fstream>
+#include <string>
+#include <cstring>
+#include <vector>
+#include <event.h>
+#include <chunkattr.h>
+#include <storywriter.h>
+
+#define DATASET_RANK 1 // Dataset dimension
+#define ATTRIBUTE_CHUNKMETADATA "ChunkMetadata"
+#define ATTRIBUTE_DATASET_MIN "Min"
+#define ATTRIBUTE_DATASET_MAX "Max"
+#define ATTRIBUTE_TOTAL_DATASET_STORY_EVENTS "TotalStoryEvents"
+#define ATTRIBUTE_CHUNKMETADATA_RANK 1
+#define DEBUG 0 // Set to 1 ito print H5 error messages to console
+#define CHUNK_SIZE 40000 // Number of events equal to page size of 4MB
+
+// Constructor storywriter
+storywriter::storywriter(){}
+
+// Destructor storywriter
+storywriter::~storywriter(){}
+
+/**
+ * Write/Append data to storyDataset. 
+ * @param storyChunk: Data to write to story dataset.
+ * @param STORY: story name
+ * @param CHRONICLE: chronicle name
+ * @return: 0 if successful, else -1, if failed.
+ */
+int storywriter::writeStoryChunk(std::vector<Event> storyChunk, const char* STORY, const char* CHRONICLE) {
```

Comment:
Maybe you should declare storyChunk as a pointer to STL vector
std::vector<Event> *storyChunk. RPC calls from the client will copy data to a memory location and the starting address of that memory location should be the input to the write API.

Reply:
Done